filebeat/harvester/log.go

```diff
@@ -104,24 +114,54 @@ func (h *Harvester) Harvest() {
 		// Reset Backoff
 		h.backoff = h.Config.BackoffDuration
 
-		// Sends text to spooler
-		event := &input.FileEvent{
-			ReadTime:     lastReadTime,
-			Source:       &h.Path,
-			InputType:    h.Config.InputType,
-			DocumentType: h.Config.DocumentType,
-			Offset:       h.Offset,
-			Bytes:        bytesRead,
-			Text:         &text,
-			Fields:       &h.Config.Fields,
-			Fileinfo:     &info,
-		}
+		if h.shouldExportLine(text) {
+
+			// Sends text to spooler
+			event := &input.FileEvent{
+				ReadTime:     lastReadTime,
+				Source:       &h.Path,
+				InputType:    h.Config.InputType,
+				DocumentType: h.Config.DocumentType,
+				Offset:       h.Offset,
+				Bytes:        bytesRead,
+				Text:         &text,
+				Fields:       &h.Config.Fields,
+				Fileinfo:     &info,
+			}
+
+			// Set Offset
+			h.Offset += int64(bytesRead) // Update offset if complete line has been processed
 
-		h.Offset += int64(bytesRead) // Update offset if complete line has been processed
+			event.SetFieldsUnderRoot(h.Config.FieldsUnderRoot)
+			h.SpoolerChan <- event // ship the new event downstream
+		} else {
 
-		event.SetFieldsUnderRoot(h.Config.FieldsUnderRoot)
-		h.SpoolerChan <- event // ship the new event downstream
+			// Eat line
+			h.Offset += int64(bytesRead) // Update offset if complete line has been processed
```

Comment:
yes, I had the impression that h.Offset is used in SetFieldsUnderRoot. I'll fix it. Thanks!